### PR TITLE
Load secrets from file ref env NAME__SECRETS

### PR DESCRIPTION
**Pull request by smirco-globality**
## Summary

Adds a json file loader that uses environment variable `{NAME}__SECRETS` 
This is to allow for kubernetes secrets files mounted as volumes to be loaded

### CODE REVIEW

File: microcosm/loaders/secrets_file.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+)
+
+from inflection import underscore
+
+from microcosm.metadata import Metadata
+
+
+def get_config_filename(metadata: Metadata) -> Optional[str]:
+    """
+    Derive a configuration file name from the FOO_SECRETS
```
Comment (afallou) [inline microcosm/loaders/secrets_file.py:25]: You mean `FOO__SECRETS`, right?

File: microcosm/loaders/secrets_file.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    named after the service of the form FOO_SECRETS.
+
+    """
+    config_filename = get_config_filename(metadata)
+    if config_filename is None:
+        return dict()
+
+    with open(config_filename, "r") as file_:
+        data = load_func(file_.read())
+        return dict(data)
```
Comment (afallou) [inline microcosm/loaders/secrets_file.py:50]: What's the expected format of the data in the file?

Comment (smirco-globality) [reply microcosm/loaders/secrets_file.py:50]: this matches the behavior of the normal config file loader, so the deserialization is performed by the passed in callable `load_func`, this is defined in `load_from_json_secrets_file` as `loads` from the json package

**Review by simpleigh-globality — COMMENTED** [review]
Is there any test coverage for this change?

**Comment by smirco-globality** [comment]
state of this repo is pretty roken on the develop branch - going to create a new branch and PR against master since it looks like the development methodology here has switched to that

**Pull request closed without merging**